pkg/pushsync/pushsync.go

```diff
@@ -231,40 +231,41 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 
 	// forwarding replication
 	storedChunk := false
-	if ps.warmedUp() && ps.topologyDriver.IsWithinDepth(chunkAddress) {
-
-		verifiedChunk, err := ps.validStamp(chunk, ch.Stamp)
-		if err != nil {
-			ps.metrics.InvalidStampErrors.Inc()
-			ps.logger.Warningf("pushsync: forwarder, invalid stamp for chunk %s", chunkAddress.String())
-		} else {
-			chunk = verifiedChunk
-			_, err = ps.storer.Put(ctx, storage.ModePutSync, chunk)
-			if err != nil {
-				ps.logger.Warningf("pushsync: within depth peer's attempt to store chunk failed: %v", err)
-			} else {
-				storedChunk = true
+	defer func() {
+		if !storedChunk {
+			if ps.warmedUp() && ps.topologyDriver.IsWithinDepth(chunkAddress) {
```

Comment:
sad

Reply:
I think this is a correction mechanism in itself, the in-neighborhood forwarders storing the chunk (and making it available thereby) in case of downstream peer errors

Reply:
If we are okay with storing the chunk multiple times as a forwarder node as a result of multiple attempts to sync a chunk, then the PR is good.

Reply:
its a good question whether store.Put works efficiently in that sense, whether it avoids at least write operations in case the chunk is already stored and then attempted to be stored again. 
on a higher level, with stewardship it is not guaranteeable that the chunk doesn't get reuploaded later on, so this probably needs to be handled in the store put efficiency

Reply:
no chunk will be stored twice if it already exists in the store. this is ensured by localstore